modules/drools/src/main/java/org/mule/module/drools/Drools.java

```diff
@@ -65,6 +66,9 @@ public Object createSession(Rules rules) throws Exception
 
         KnowledgeBaseConfiguration conf = 
             KnowledgeBaseFactory.newKnowledgeBaseConfiguration(null, Thread.currentThread().getContextClassLoader());
+
+        conf.setOption(AssertBehaviorOption.EQUALITY);
```

Comment:
I guess that different modes exist for a reason... woyld be good to have a way to configure drools using tje old mode (which could be better for perfoema ce reasons umder some scenarios)